### PR TITLE
chore: add "changes" filter to membership-token program CI

### LINKED ISSUE #162: membership-token program CI is running unnecessarily
membership-token program CI is being run for all PRs because it is missing the `changes` setting, which is responsible for filtering out CI jobs which do not need be run based on the files changed.

https://github.com/metaplex-foundation/metaplex-program-library/blob/master/.github/workflows/program-membership-token.yml

See auction-house for an [example](https://github.com/metaplex-foundation/metaplex-program-library/blob/master/.github/workflows/program-auction-house.yml#L15-L31) of what we need in program-membership-token.yml.

### PR DESCRIPTION
**Pull request by aheckmann**
Prevents membership-token program CI from running unless its related files change.

closes #162

### CODE REVIEW

File: .github/workflows/program-membership-token.yml [hunk 1]
```diff
@@ -12,6 +12,23 @@ env:
...
+      # For pull requests it's not necessary to checkout the code
+      - uses: dorny/paths-filter@v2
+        id: filter
+        with:
+          filters: |
+            core:
+              - 'core/**'
+            package:
+              - 'membership-token/**'
   build-and-test-membership-token:
```
Comment (thlorenz) [inline .github/workflows/program-membership-token.yml:32]: This is missing the check for the outcome of the path filter, i.e. from auction house:

```
  build-and-test-auction-house:
    needs: changes
    if: ${{ needs.changes.outputs.core == 'true' || needs.changes.outputs.package == 'true' }}
    ```

**Review by thlorenz — CHANGES_REQUESTED** [review]

File: .github/workflows/program-membership-token.yml (hunk 1 above)
Comment (aheckmann) [reply .github/workflows/program-membership-token.yml:32]: ack

**Comment by aheckmann** [comment]
Patched. All tests passing.

**Review by thlorenz — APPROVED** [review]
LGTM

**Pull request merged**